lib/phoenix/socket/router.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Phoenix.Socket.Router do
+  # Routes WebSocket and LongPoller requests.
+  @moduledoc false
+
+  use Plug.Builder
+  alias Phoenix.Transports.WebSocket
+  alias Phoenix.Transports.LongPoller
+
+  plug Plug.Logger
+  plug :fetch_query_params
+  plug :dispatch
+
+  def dispatch(conn, _) do
+    case {conn.method, transport(conn)} do
+      {"GET",  WebSocket}     -> WebSocket.call(conn, [])
+      {"POST", WebSocket}     -> WebSocket.call(conn, [])
+      {"OPTIONS", LongPoller} -> LongPoller.call(conn, :options)
+      {"GET", LongPoller}     -> LongPoller.call(conn, :poll)
+      {"POST", LongPoller}    -> LongPoller.call(conn, :publish)
+      _                       -> conn |> send_resp(:bad_request, "") |> halt()
+    end
+  end
+
+  defp transport(conn) do
+    case conn.query_params["transport"] do
```

Comment:
I had that at first, but stomping the entire socket mount namespace seems wrong. So we could hardcode `["ws", "poll"]`, with an extra clause, but this also felt wrong. So I re-evaluated why we needed an extra path, and I think it makes more since to have a single socket mount point and the client tells us which transport it wants. It is a breaking change for long polling clients (they need new js), but old ws clients would still work, which I am okay with. Better?